### PR TITLE
5.4 new mentor page registration pitsyk

### PR DESCRIPTION
**Pull request by staspitsyk**
Created teachers-registration-form component
Added notifications

### CODE REVIEW

File: front/src/common/redux/root-saga.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import { fork } from 'redux-saga/effects';
-import { studentsSaga } from './students/students.saga';
+import {studentsSaga} from './students/students.saga';
```
Comment (so2niko) [inline front/src/common/redux/root-saga.js]: add spaces

File: front/src/common/redux/root-saga.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
 import { authSaga } from './auth/auth.saga';
+import {teachersSaga} from './teachers/teachers.saga';
```
Comment (so2niko) [inline front/src/common/redux/root-saga.js]: spaces

File: front/src/common/redux/teachers/teachers.actions.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { SHOW_TEACHER_REGISTER_MODAL, HIDE_TEACHER_REGISTER_MODAL, REGISTER_TEACHER, SHOW_LOADER } from './types';
```
Comment (so2niko) [inline front/src/common/redux/teachers/teachers.actions.js]: wrap to different lines in accordance with the general style of the code

File: front/src/common/redux/teachers/teachers.actions.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+export const registerTeacher = (teacher) => ({type: REGISTER_TEACHER,payload: teacher});
+
+export const hideTeacherRegisteredModal = () => ({type: HIDE_TEACHER_REGISTER_MODAL})
```
Comment (so2niko) [inline front/src/common/redux/teachers/teachers.actions.js]: semicolon

File: front/src/common/redux/teachers/teachers.actions.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+
+export const showTeacherRegisteredModal = () => ( {type: SHOW_TEACHER_REGISTER_MODAL})
```
Comment (so2niko) [inline front/src/common/redux/teachers/teachers.actions.js]: semicolon

File: front/src/common/redux/teachers/teachers.actions.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+
+export const showLoader = () => ({type: SHOW_LOADER})
```
Comment (so2niko) [inline front/src/common/redux/teachers/teachers.actions.js]: semicolon

**Review by so2niko — COMMENTED** [review]
some styles marks

File: front/src/common/redux/root-saga.js (hunk 1 above)
Comment (staspitsyk) [reply front/src/common/redux/root-saga.js]: Done

File: front/src/common/redux/root-saga.js (hunk 2 above)
Comment (staspitsyk) [reply front/src/common/redux/root-saga.js]: Done

File: front/src/common/redux/teachers/teachers.actions.js (hunk 3 above)
Comment (staspitsyk) [reply front/src/common/redux/teachers/teachers.actions.js]: Done

File: front/src/common/redux/teachers/teachers.actions.js (hunk 4 above)
Comment (staspitsyk) [reply front/src/common/redux/teachers/teachers.actions.js]: Done

File: front/src/common/redux/teachers/teachers.actions.js (hunk 5 above)
Comment (staspitsyk) [reply front/src/common/redux/teachers/teachers.actions.js]: Done

File: front/src/common/redux/teachers/teachers.actions.js (hunk 6 above)
Comment (staspitsyk) [reply front/src/common/redux/teachers/teachers.actions.js]: Done

**Review by so2niko — APPROVED** [review]
All is OK

**Review by maslovakat — APPROVED** [review]

**Review by antonpanchenk0 — APPROVED** [review]

File: front/src/pages/teachers/teachers-registration-form/index.js [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+                  <Input placeholder="Email" />
+                </Form.Item>
+                <Form.Item
+                  name="password"
+                  align="center"
+                  rules={[
+                    { required: true, message: 'Please input your password!' },
+                    {
+                      validator: (_, value) =>
+                        value.length >= 6
```
Comment (Alan18081) [inline front/src/pages/teachers/teachers-registration-form/index.js]: Please, move it to the validators file and make it in the way that it takes `6` as an argument

File: front/src/components/app/index.js [hunk 8]
```diff
@@ -9,11 +12,17 @@ class App extends Component {
...
                     <Route path="/login" component={LoginPage} />
                 </Switch>
             </BrowserRouter>
         )
     }
 }
 
-export default App;
\ No newline at end of file
+export default connect()(App);
```
Comment (Alan18081) [inline front/src/components/app/index.js]: Why do we need `connect` here without arguments

File: front/src/common/redux/teachers/teachers.saga.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    showNotification(
+      'Successfully registered',
+      `User with provided email ${response.data.email} was registered`,
+      'success'
+    );
+
+    yield put({ type: TEACHER_COMPLETED_REGISTER });
+  } catch (error) {
+    if (error.response) {
+      showNotification('Sorry but you did not register', error.response.data.error, 'error');
```
Comment (Alan18081) [inline front/src/common/redux/teachers/teachers.saga.js]: It is better to change the title to something like `Registration error`

File: front/src/common/redux/teachers/teachers.saga.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+      yield put({ type: TEACHER_REGISTER_ERROR_HAPPENED });
+    } else {
+      showNotification('Sorry but you did not register', error.message, 'error');
```
Comment (Alan18081) [inline front/src/common/redux/teachers/teachers.saga.js]: It is better to change the title to something like `Registration error`

File: front/src/components/app/index.js (hunk 8 above)
Comment (staspitsyk) [reply front/src/components/app/index.js]: Added mapDispatchToProps

File: front/src/pages/teachers/teachers-registration-form/index.js (hunk 7 above)
Comment (staspitsyk) [reply front/src/pages/teachers/teachers-registration-form/index.js]: Removed inline validator function, instead passed 6 as minLength argument to Input.Password

File: front/src/common/redux/teachers/teachers.saga.js (hunk 9 above)
Comment (staspitsyk) [reply front/src/common/redux/teachers/teachers.saga.js]: Done

File: front/src/common/redux/teachers/teachers.saga.js (hunk 10 above)
Comment (staspitsyk) [reply front/src/common/redux/teachers/teachers.saga.js]: Done

File: front/src/components/app/index.js (hunk 8 above)
Comment (staspitsyk) [reply front/src/components/app/index.js]: I removed this logic at all. It crashes the app in case of adding some queries for pagination to URL

**Review by Alan18081 — APPROVED** [review]

**Pull request merged**